internal/dataplane/parser/translate_tcproute.go

```diff
@@ -103,12 +103,6 @@ func generateKongRoutesFromTCPRouteRule(
 	objectInfo := util.FromK8sObject(tcproute)
 
 	var routes []kongstate.Route
-	if len(rule.Matches) > 0 {
-		// As of 2022-03-04, matches are supported only in experimental CRDs. if you apply a TCPRoute with matches against
-		// the stable CRDs, the matches disappear into the ether (only if doing it via client-go, kubectl rejects them)
-		// We do not intend to implement these until they are stable per https://github.com/Kong/kubernetes-ingress-controller/issues/2087#issuecomment-1079053290
-		return routes, fmt.Errorf("TCPRoute Matches are not yet supported")
-	}
```

Comment:
Do we not need to add an actual implementation for handling Matches as well? They may now be stable but these errors are still correct on our end AFAIK.